### PR TITLE
OCPBUGS-9714: Any namespace after deletion is still visible on UI if it is the default selection in namespace dropdown

### PR DESCRIPTION
**Pull request by Lucifergene**
**Fixes**: 
https://issues.redhat.com/browse/OCPBUGS-9714

**Analysis / Root cause**: 

- [x] Any namespace after deletion is still visible on UI if it is the default selection in the namespace dropdown.
- [x] If there is an ns mentioned in the URL, that does not exist, it is still shown in the namespace dropdown.

**Solution Description**: 
I have added a check in the Projects Dropdown to ensure the activeNamespace is present in the cluster. If not, the dropdown autoselects "All Projects" and shows the Project List.

_Note: If you delete an ns from a different tab or via CLI, ensure to refresh the actual page._

**Screen shots / Gifs for design review**: 

https://github.com/openshift/console/assets/47265560/47ce3e66-bb9c-41b7-a3a6-e385ad2b2a52

Normal User:

https://github.com/openshift/console/assets/47265560/77653ccf-325e-4d70-861c-867c22c7bae9

**Unit test coverage report**: 
Not Changed

**Test setup:**
1. Open 2 tabs: 
      a. Namespace in Admin View - TAB 1
      b. Topology Page in Dev View - TAB 2
2. Create a ns in the TAB 1.
3. In TAB 2, select that ns in the dropdown.
4. In TAB 1, delete the ns.
5. In TAB 2, refresh the page. You will see that the Dropdown is changed to "All Projects" and the Project list is shown.
6. In TAB 2, select some other project to verify that the ns is changed in the Project dropdown.
7. Now in the TAB 2 URL, change the namespace to something else that does not exist and press ENTER.
8. You will see that the Project Dropdown changes to "All Projects" and you see the Project List again.

**Browser conformance**: 

- [x] Chrome
- [ ] Firefox
- [ ] Safari
- [ ] Edge

### CODE REVIEW

**Review by jerolimov — CHANGES_REQUESTED** [review]

File: frontend/public/components/namespace-bar.tsx [hunk 1]
```diff
@@ -41,12 +41,20 @@ export const NamespaceBarDropdowns: React.FC<NamespaceBarDropdownsProps> = ({
...
     if (namespace.loaded) {
       dispatch(setFlag(FLAGS.SHOW_OPENSHIFT_START_GUIDE, _.isEmpty(namespace.data)));
     }
   }, [dispatch, namespace.data, namespace.loaded]);
+  React.useEffect(() => {
+    if (nsErr) {
+      /* This would redirect to "/all-namespaces" to show the Project List */
+      setActiveNamespace(ALL_NAMESPACES_KEY);
+    }
+  }, [nsErr]);
```
[the remark this answers is not in the document]
Comment (Lucifergene) [reply frontend/public/components/namespace-bar.tsx]: Thanks. Have updated the code.

**Comment by vikram-raj** [comment]
@Lucifergene e2e is failing because of the change in PR. Please take a look.

File: frontend/packages/integration-tests-cypress/tests/crud/namespace-crud.cy.ts [hunk 2]
```diff
@@ -77,7 +77,6 @@ describe('Namespace', () => {
     projectDropdown.shouldContain(defaultProjectName);
     detailsPage.breadcrumb(0).contains('Services').click();
     listPage.rows.shouldBeLoaded();
-    projectDropdown.shouldContain(allProjectsDropdownLabel);
```
Comment (Lucifergene) [inline frontend/packages/integration-tests-cypress/tests/crud/namespace-crud.cy.ts:80]: When we click on a list item with a namespace set in the URL, the project dropdown will automatically select that ns, as it continuously monitors the ns in the URL. 
So, I've removed this line from the test.

**Comment by vikram-raj** [comment]
@Lucifergene still one e2e test is failing.

`Nav and breadcrumbs restores last selected "All Projects" when navigating from details to list view: Namespace Nav and breadcrumbs restores last selected "All Projects" when navigating from details to list view `

**Review by vikram-raj — COMMENTED** [review]
Tested it on the cluster bot and noticed an issue. All project is selected and when I create a new project it should select the new project but it is not get selected at once. See the recording.

https://github.com/openshift/console/assets/2561818/8b2dde67-ba58-4e4a-b54f-83fc3c8d7f8e

**Comment by Lucifergene** [comment]
@vikram-raj This seems a legitimate issue as it takes time to get created on the cluster when you create the new namespace. But the console does not wait for that and directly changes the URL to the new ns. But since it's not created initially, the All Projects gets shown and after some time, when the ns is finally created in the cluster, the dropdown changes to it.

I am trying to approach this so that when you create the namespace, it waits for it to get created and then changes the route.

**Review by vikram-raj — COMMENTED** [review]
/lgtm

**Pull request merged**